DigitalLearningSolutions.Web/Views/MyAccount/_EditRegistrationNumber.cshtml

```diff
@@ -0,0 +1,67 @@
+﻿@using DigitalLearningSolutions.Web.Models.Enums
+@using DigitalLearningSolutions.Web.ViewModels.MyAccount
+@using Microsoft.AspNetCore.Mvc.ModelBinding
+
+@model MyAccountEditDetailsViewModel
+
+@{
+  var errorHasOccurred = false;
+  if (ViewData.ModelState.TryGetValue(nameof(Model.HasProfessionalRegistrationNumber), out var modelState)) {
+    errorHasOccurred = modelState.ValidationState != ModelValidationState.Valid;
+  }
+
+  var optionYesSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.Yes;
+  var optionNoSelected = Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.No;
+  const string professionalRegConditionalId = "professional-reg-conditional-Id";
+}
+
+<div class="nhsuk-form-group @(errorHasOccurred ? "nhsuk-form-group--error" : "")">
+  <fieldset class="nhsuk-fieldset" aria-describedby="professional-registration-hint">
+    <div class="nhsuk-hint" id="professional-registration-hint">
+      Do you have a Professional Registration Number?
+    </div>
+
+    @if (errorHasOccurred) {
+      <span class="nhsuk-error-message" id="no-selection-error">
+        <span class="nhsuk-u-visually-hidden">Error:</span> @ViewData.ModelState[nameof(Model.HasProfessionalRegistrationNumber)].Errors[0].ErrorMessage
+      </span>
+    }
+
+    <div class="nhsuk-radios nhsuk-radios--conditional">

```

Comment:
This is related to the point I raised in slack regarding the radio implementation.
We have a view component used for radios in `EditAdminRoles.cshtml`. 
Whether that implementation is overly complicated for this scenario or not is a question. But as you point out the dependent expanders add complexity. 
There is certainly scope to create a component.

Reply:
Some of the challenges here which may over complicate matters include:
- The disconnect between the radios and associated expandables and the desire to ensure clarity on the relationship between the two
- A solution to encompasses the expanded content looks overly complex, especially given the expandable content varies.

Reply:
I had a sense it would be hard to commonise.
You could pull this out into a PRN-specific partial to share with registration if it seems convenient?

Reply:
Something needs to be done. I took a look at this but the way it was originally put together means it is slightly more complex.
The view takes a specific model for editing the account details so as it stands does not work as a shared view given registration is a different model. This needs more work.

Reply:
Sure, let's look at it again in the Registration ticket. Would an interface `IIncludesPrnForm`, implemented by the two form data classes, be the solution?

Reply:
@SteveJacksonSoft that was my initial though. It should be very straight forward. It's kind of a workaround but the lesser of the two evils as it will avoid more complex refactor.